@here/harp-mapview-decoder/lib/TileDataSource.ts

```diff
@@ -201,18 +208,74 @@ export class TileDataSource<TileType extends Tile> extends DataSource {
     getTile(tileKey: TileKey): TileType | undefined {
         const tile = this.m_tileFactory.create(this, tileKey);
 
-        tile.tileLoader = new TileLoader(
-            this,
-            tileKey,
-            this.m_options.dataProvider,
-            this.decoder,
-            0
-        );
+        const mortonCode = tileKey.mortonCode();
+        const tileLoader = this.m_tileLoaderCache.get(mortonCode);
+        // Filter out disposed TileLoaders.
+        if (tileLoader !== undefined && tileLoader.state !== TileLoaderState.Disposed) {
+            tile.tileLoader = tileLoader;
```

Comment:
Shouldn't we remove the disposed ones from the cache immediately?

Reply:
Unfortunately it isn't currently possible, the Tile::dispose method calls TileLoader::dispose, however TileLoader doesn't have a reference to the TileDataSource, just to the DataSource (which contains owns the TileLoaders). I could add it, but it would make the code a bit more ugly, because it would involve type checking etc.

Also, I found a bug, currently the TileLoader.dispose method doesn't check correctly if it can actually dispose, or whether there is another Tile waiting on it. I will fix and add test.

Actually... I don't think it makes sense to have a Disposed state, because it anyway just calls cancel, and then sets the disposed state. A disposed TileLoader can't be restarted like a cancelled one, but I don't quite see the use case for this.